src/fr-policy/index.ts

```diff
@@ -0,0 +1,109 @@
+// {
+//   "code": 403,
+//   "reason": "Forbidden",
+//   "message": "Policy validation failed",
+//   "detail": {
+//     "result": false,
+//     "failedPolicyRequirements": [
+//       {
+//         "policyRequirements": [{ "policyRequirement": "UNIQUE" }],
+//         "property": "userName"
+//       }
+//     ]
+//   }
+// }
+
+import { Step } from 'auth/interfaces';
+import { FailedPolicyRequirement, PolicyRequirement } from '../auth/interfaces';
+import { PolicyMessage, PolicyName } from './enums';
+import { ProcessedPropertyError } from './interfaces';
+
+abstract class FRPolicy {
+
+  public static parseErrors(err: Partial<Step>): ProcessedPropertyError[] {
+    let errors: ProcessedPropertyError[] = [];
+    if (err.detail && err.detail.failedPolicyRequirements) {
+      errors = [];
+      err.detail.failedPolicyRequirements.map((x) => {
+        errors.push.apply(errors, [{ messages: this.parseFailedPolicyRequirement(x), detail: x}]);
+      });
+    }
+    return errors;
+  }
+
+ public static parseFailedPolicyRequirement(policy: FailedPolicyRequirement): string[] {
+    const errors: string[] = [];
+    policy.policyRequirements.map((x) => {
+      errors.push(this.parsePolicyRequirement(policy.property, x));
+    });
+    return errors;
+  }
+
+  public static parsePolicyRequirement(property: string, policy: PolicyRequirement): string {
+
+    // Object.keys(PolicyName).forEach((name) => {
+    //   if (PolicyName[name] === policy.policyRequirement) {
+
+    //   }
+    // });
+
+    // for (const key in PolicyName) {
+    //   if ( PolicyName[key as PolicyName]  === policy.policyRequirement) {
+
+    //   }
+    // }
+    switch (policy.policyRequirement) {
```

Comment:
this is a huge and not pretty switch statement but trying to iterate through the enum gave me TS type errors that couldn't be fixed